core/modules/filter/filter.admin.inc

```diff
@@ -287,35 +292,42 @@ function filter_admin_format_form($form, &$form_state, $format) {
   }
   $form['#filters'] = $filters;
 
-  // Filter status.
-  $form['filters']['status'] = array(
+  // Filter order (tabledrag).
+  $form['filters'] = array(
     '#type' => 'item',
-    '#title' => t('Enabled filters'),
-    '#prefix' => '<div id="filters-status-wrapper">',
-    '#suffix' => '</div>',
+    '#title' => t('Filter settings'),
+    '#theme' => 'filter_admin_format_filter_order',
+  );
+  $form['filters']['filters'] = array(
+    '#type' => 'value',
+    '#value' => $filters,
   );
   foreach ($all_filter_info as $name => $filter_info) {
-    $form['filters']['status'][$name] = array(
+    $configure_form = _filter_build_configure_form($form, $form_state, $name, $format);
+    $configure_link = !empty($configure_form) ? l(t('Configure'), '#', array('attributes' => array('class' => 'configure-link'), 'fragment' => 'configure')) : '';
+    
+    $form['filters'][$name]['status'] = array(
       '#type' => 'checkbox',
-      '#title' => $filter_info['title'],
       '#default_value' => $filters[$name]->status,
-      '#parents' => array('filters', $name, 'status'),
-      '#description' => $filter_info['description'],
-      '#weight' => $filter_info['weight'],
+      '#attributes' => array('class' => array('filter-status')),
     );
-  }
-
-  // Filter order (tabledrag).
-  $form['filters']['order'] = array(
-    '#type' => 'item',
-    '#title' => t('Filter processing order'),
-    '#theme' => 'filter_admin_format_filter_order',
-  );
-  foreach ($all_filter_info as $name => $filter_info) {
-    $form['filters']['order'][$name]['filter'] = array(
+    $form['filters'][$name]['configure_link'] = array(
+      '#markup' => $configure_link,
+    );
+    $form['filters'][$name]['filter'] = array(
+      '#theme' => 'filter_admin_format_filter_display',
+      '#type' => 'item',
+      '#prefix' => '<div class="filter-td">',
+      '#suffix' => '</div>',
+    );
+    $form['filters'][$name]['filter']['title'] = array(
       '#markup' => $filter_info['title'],
     );
-    $form['filters']['order'][$name]['weight'] = array(
+    $form['filters'][$name]['filter']['details'] = array(
+      '#markup' => isset($format_tips[$name]['tip']) ? $format_tips[$name]['tip'] : '',
+    );
+    $form['filters'][$name]['filter']['configure'] = $configure_form;
```

Comment:
This might be a good place to add an AJAX button to validate the form and close the dialog.

```php
$configure_form['actions']['#type'] = 'actions';
$configure_form['actions']['submit'] = array(
  '#type' => 'submit',
  '#value' => t('Update settings'),
  '#limit_validation_errors' => array($configure_form['#parents']),
  '#ajax' => array(
      'event' => 'click',
      'callback' => 'filter_admin_filter_form_ajax_update',
      'progress' => array('type' => 'throbber', 'message' => NULL),
  ),
  '#submit' => array('filter_admin_filter_form_submit'),
);
```

Then using `filter_admin_filter_form_submit`, have it rebuild the form. And in `filter_admin_filter_form_ajax_update`, have it pull out the parts of the form that need to be updated and then close the dialog. This is probably going to look a lot like the existing [layout_ajax_form_save_dialog](https://api.backdropcms.org/api/backdrop/core!modules!layout!layout.admin.inc/function/layout_ajax_form_save_dialog/1) function.

Unfortunately, AJAX saving and form updating is not easy. This will make the PR quite a bit more complicated.